step-templates/hashicorp-vault-approle-login.json

```diff
@@ -0,0 +1,75 @@
 not specified\"\n}\nif ([string]::IsNullOrWhiteSpace($VAULT_LOGIN_API_VERSION)) {\n\tthrow \"Required parameter VAULT_LOGIN_API_VERSION not specified\"\n}\nif ([string]::IsNullOrWhiteSpace($VAULT_LOGIN_APPROLE_PATH)) {\n\tthrow \"Required parameter VAULT_LOGIN_APPROLE_PATH not specified\"\n}\nif ([string]::IsNullOrWhiteSpace($VAULT_LOGIN_ROLEID)) {\n\tthrow \"Required parameter VAULT_LOGIN_ROLEID not specified\"\n}\nif ([string]::IsNullOrWhiteSpace($VAULT_LOGIN_SECRETID)) {\n\tthrow \"Required parameter VAULT_LOGIN_SECRETID not specified\"\n}\n\n# Helper functions\n###############################################################################\nfunction Get-WebRequestErrorBody {\n    param (\n        $RequestError\n    )\n\n    # Powershell < 6 you can read the Exception\n    if ($PSVersionTable.PSVersion.Major -lt 6) {\n        if ($RequestError.Exception.Response) {\n            $reader = New-Object System.IO.StreamReader($RequestError.Exception.Response.GetResponseStream())\n            $reader.BaseStream.Position = 0\n            $reader.DiscardBufferedData()\n            $response = $reader.ReadToEnd()\n\n            return $response | ConvertFrom-Json\n        }\n    }\n    else {\n        return $RequestError.ErrorDetails.Message\n    }\n}\n###############################################################################\n\n$VAULT_LOGIN_ADDRESS = $VAULT_LOGIN_ADDRESS.TrimEnd('/')\n$VAULT_LOGIN_APPROLE_PATH = $VAULT_LOGIN_APPROLE_PATH.TrimStart('/').TrimEnd('/')\n\n# Local variables\n$StepName = $OctopusParameters[\"Octopus.Step.Name\"]\n\ntry\n{\n\t$payload = @{\n    \trole_id = $VAULT_LOGIN_ROLEID\n        secret_id = $VAULT_LOGIN_SECRETID\n    }\n    \n    $uri = \"$VAULT_LOGIN_ADDRESS/$VAULT_LOGIN_API_VERSION/auth/$VAULT_LOGIN_APPROLE_PATH/login\"\n    $headers = @{ }\n    Write-Verbose \"Making request to $uri\"\n    $response = Invoke-RestMethod -Method Post -Uri $uri -Body ($payload | ConvertTo-Json -Depth 10) -Headers $headers\n    \n    if($null -ne $response) {\n    \tWrite-Host \"Setting client token output variable.\"\n        Write-Verbose \"Access via '`$OctopusParameters[Octopus.Action[$StepName].Output.AppRoleAuthToken]' or '##{Octopus.Action[$StepName].Output.AppRoleAuthToken}'\"\n    \tSet-OctopusVariable -Name \"AppRoleAuthToken\" -Value $response.auth.client_token -Sensitive\n    }\n    else {\n    \tWrite-Error \"Null or Empty response returned from Vault server\" -Category InvalidResult\n    }\n}\ncatch\n{\n\t$ExceptionMessage = $_.Exception.Message\n\t$ErrorBody = Get-WebRequestErrorBody -RequestError $_\n    $Message = \"An error occurred logging in with AppRole: $ExceptionMessage\"\n    if($null -ne $ErrorBody.errors) {\n    \t$AdditionalDetail += $ErrorBody.errors -Join \",\"\n    }\n    else {\n    \t$AdditionalDetail += $ErrorDetails\n    }\n    if (![string]::IsNullOrWhiteSpace($AdditionalDetail)) {\n    \t$Message += \"`n`tDetail: $AdditionalDetail\"\n\t}\n\n    Write-Error $Message -Category ConnectionError\n}"
```

Comment:
Since the script is setting an output variable could you do a write-highlight and let the person know the name of the output variable?  You don't have to tell them the value, only that it is being set.

Reply:
Added in https://github.com/OctopusDeploy/Library/commit/517ee0900d37c8c4475fc1ec8e90592bebbceaab